### PR TITLE
Backups: Switch pgzip -> pargzip for lower memory consumption

### PR DESCRIPTION
**Pull request by teejae**
This solved #5613 for us.

Signed-off-by: Toliver Jue <toliver@planetscale.com>

### CODE REVIEW

File: go/vt/mysqlctl/xtrabackupengine.go [hunk 1]
```diff
@@ -269,19 +270,18 @@ func (be *XtrabackupEngine) backupFiles(ctx context.Context, params BackupParams
...
 		if *backupStorageCompress {
-			compressor, err := pgzip.NewWriterLevel(writer, pgzip.BestSpeed)
-			if err != nil {
-				return replicationPosition, vterrors.Wrap(err, "cannot create gzip compressor")
-			}
-			compressor.SetConcurrency(*backupCompressBlockSize, *backupCompressBlocks)
+			compressor := pargzip.NewWriter(writer)
+			compressor.ChunkSize = *backupCompressBlockSize
+			compressor.Parallel = *backupCompressBlocks
+			compressor.CompressionLevel = pargzip.BestCompression
```
Comment (enisoc) [inline go/vt/mysqlctl/xtrabackupengine.go:284]: Did you mean BestSpeed?

Comment (teejae) [reply go/vt/mysqlctl/xtrabackupengine.go]: yes, pushed new version. thanks for the catch.

**Comment by teejae** [comment]
both @systay and myself have tested `master` and this branch and see the failed test. i don't believe the test failure is related to this branch.

**Comment by deepthi** [comment]
> both @systay and myself have tested `master` and this branch and see the failed test. i don't believe the test failure is related to this branch.

We are ok with merging if a "flaky" test fails that isn't related to the changes. Since the failing test _is_ related, we should get it passing at least once before merge.

**Comment by teejae** [comment]
@deepthi i'm now not so confident in the PR. i can't figure out what's wrong, but the fact that the test never finishes is not a good sign.

**Comment by harshit-gangal** [comment]
I see the test failure as

```
F0716 06:51:32.116305  196200 tm_init.go:654] RestoreFromBackup failed: can't restore file 10 to mysql/general_log.CSV: can't open gzip decompressor: EOF
failed to restore files
Can't restore backup
```

**Comment by harshit-gangal** [comment]
Should look at https://github.com/vitessio/vitess/blob/master/go/vt/mysqlctl/builtinbackupengine.go#L559,L575

```
gz, err := gzip.NewReader(reader)
if err != nil {
	return vterrors.Wrap(err, "can't open gzip decompressor")
}
```

**Comment by systay** [comment]
Re-open if/when we want to merge this

**Pull request closed without merging**